trading_calendars/exchange_calendar_xbue.py

```diff
@@ -0,0 +1,373 @@
+    previous Monday.  If it falls on Thursday, Friday or a weekend,
+    it is observed on the following Monday.
+    """
+    day = dt.weekday()
+
+    if day in (MONDAY, TUESDAY, WEDNESDAY):
+        return dt - timedelta(day)
+
+    while dt.weekday() != MONDAY:
+        dt += timedelta(1)
+
+    return dt
+
+
+def cultural_diversity_observance(holidays):
+    """
+    Apply the nearest monday rule, and also exclude 2012 (Day of
+    Respect for Cultural Diversity breaks the nearest monday rule
+    in 2012).
+    """
+    holidays = pd.to_datetime(
+        holidays.map(lambda d: nearest_monday(d))
+    )
+    return holidays[holidays.year != 2012]
+
+
+def not_2018(holidays):
+    """
+    Exclude the year 2018 from the holiday list.
+    """
+    return holidays[holidays.year != 2018]
+
+
+NewYearsDay = new_years_day()
+
+CarnivalMonday = Holiday(
+    'Carnival Monday',
+    month=1,
+    day=1,
+    start_date='2011',
+    offset=[Easter(), -Day(48)],
+)
+
+CarnivalTuesday = Holiday(
+    'Carnival Tuesday',
+    month=1,
+    day=1,
+    start_date='2011',
+    offset=[Easter(), -Day(47)],
+)
+
+TruthAndJusticeMemorialDay = Holiday(
+    'Truth and Justice Memorial Day',
+    month=3,
+    day=24,
+    start_date='2005',
+)
+
+MalvinasDayTo2004 = Holiday(
+    'Malvinas Day (up to 2004)',
+    month=3,
+    day=31,
+    end_date='2005',
+    offset=pd.DateOffset(weekday=MO(1)),
+)
+
+MalvinasDayFrom2005 = Holiday(
+    'Malvinas Day (2005-present)',
+    month=4,
+    day=2,
+    start_date='2005'
+)
+
+MaundyThursday = maundy_thursday()
+
+LabourDay = european_labour_day()
+
+MayDayRevolution = Holiday(
+    'May Day Revolution',
+    month=5,
+    day=25,
+)
+
+MartinMiguelDeGuemesDay = Holiday(
+    'Martin Miguel de Guemes Day',
+    month=6,
+    day=17,
+    start_date='2016',
+)
+
+NationalFlagDayTo2010 = Holiday(
+    'National Flag Day (up to 2010)',
+    month=6,
+    day=1,
+    end_date='2011',
+    offset=pd.DateOffset(weekday=MO(3)),
+)
+
+NationalFlagDayFrom2011 = Holiday(
+    'National Flag Day (2011-present)',
+    month=6,
+    day=20,
+    start_date='2011',
+)
+
+IndependenceDay = Holiday(
+    'Independence Day',
+    month=7,
+    day=9,
+)
+
+SanMartinsDayTo2010 = Holiday(
+    "San Martin's Day",
+    month=8,
+    day=1,
+    end_date='2011',
+    offset=pd.DateOffset(weekday=MO(3)),
+)
+
+SanMartinsDayFrom2012 = Holiday(
+    "San Martin's Day",
+    month=8,
+    day=1,
+    start_date='2012',
+    offset=pd.DateOffset(weekday=MO(3)),
+)
+
+DayOfRespectForCulturalDiversity = Holiday(
+    "Day of Respect for Cultural Diversity",
+    month=10,
+    day=12,
+    observance=cultural_diversity_observance,
+)
+
+BankHoliday = Holiday(
+    'Bank Holiday',
+    month=11,
+    day=6,
+    observance=not_2018,
+)
+
+DayOfNationalSovereigntyTo2016 = Holiday(
+    'Day of National Sovereignty (2010-2016)',
+    month=11,
+    day=1,
+    start_date='2010',
+    end_date='2015',
```

Comment:
If it's `To2016` should this be '2017'?

Reply:
You're right, I think I forgot to rename this at some point!